src/rgw/rgw_swift_auth.cc

```diff
@@ -524,8 +524,10 @@ void RGW_SWIFT_Auth_Get::execute()
       protocol = "https";
     } else {
       server_port = s->info.env->get("SERVER_PORT");
-      add_port = (strcmp(server_port, "80") != 0);
-      protocol = "http";
+      if (server_port) {
```

Comment:
this is not addressed, we should have a default value for protocol. what if neither "SERVER_PORT_SECURE" nor "SERVER_PORT" is specified?

`swift_url = protocol;` at https://github.com/ceph/ceph/pull/13293/commits/06b7d5c9ef6806b3bd1929f26d58382251ff1701#diff-e99d9bc966465b3d31f9909075b6ecc8L536 will crash.